src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectionRuleOptionsTests.cs

```diff
@@ -829,6 +829,156 @@ public Task CollectionRuleOptions_LoadProfilerAction_ClsidPropertyValidation()
+                },
+                ruleOptions =>
+                {
+                    Assert.Single(ruleOptions.Actions);
+                    ruleOptions.VerifySetEnvironmentVariableAction(0, VariableName, VariableValue);
+                });
+        }
+
+        [Fact]
+        public async Task CollectionRuleOptions_SetEnvironmentVariable_NamePropertyValidation()
+        {
+            await ValidateFailure(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction(null);
+                },
+                ex =>
+                {
+                    string[] failures = ex.Failures.ToArray();
+                    Assert.Single(failures);
+                    VerifyRequiredMessage(failures, 0, nameof(SetEnvironmentVariableOptions.Name));
+                });
+
+            await ValidateFailure(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction(string.Empty);
+                },
+                ex =>
+                {
+                    string[] failures = ex.Failures.ToArray();
+                    Assert.Single(failures);
+                    VerifyRequiredMessage(failures, 0, nameof(SetEnvironmentVariableOptions.Name));
+                });
+
+            await ValidateFailure(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction("    ");
+                },
+                ex =>
+                {
+                    string[] failures = ex.Failures.ToArray();
+                    Assert.Single(failures);
+                    VerifyRequiredMessage(failures, 0, nameof(SetEnvironmentVariableOptions.Name));
+                });
+        }
+
+        [Fact]
+        public Task CollectionRuleOptions_GetEnvironmentVariable_RoundTrip()
+        {
+            const string VariableName = "MyGettingVar";
+
+            return ValidateSuccess(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddGetEnvironmentVariableAction(VariableName);
+                },
+                ruleOptions =>
+                {
+                    Assert.Single(ruleOptions.Actions);
+                    ruleOptions.VerifyGetEnvironmentVariableAction(0, VariableName);
+                });
+        }
+
+        [Fact]
+        public async Task CollectionRuleOptions_GetEnvironmentVariable_NamePropertyValidation()
+        {
+            await ValidateFailure(
```

Comment:
Same comment as above -> consider converting to theory with the environment variable name passed in as a parameter.